R-package/configure.win

```diff
@@ -70,6 +70,34 @@ then
     LGB_CPPFLAGS="${LGB_CPPFLAGS} -DMM_MALLOC=1"
 fi
 
+###############
+# INET_PTON #
+###############
+
+ac_inet_pton="no"
+
+cat > conftest.cpp <<EOL
+#ifdef _WIN32
```

Comment:
@jameslamb Do we need this guard here given that this test is located in configure.**win**. I guess this file is run only on Windows.

Reply:
good point! I think you're right, guards should be removed here to avoid false negatives.

Reply:
For windows, there's also a macro `_WIN64 `. I'm not sure whether `_WIN32` covers all windows systems.

Reply:
Hmmm, you're right!

> `_WIN32` Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.

> `_WIN64` Defined as 1 when the compilation target is 64-bit ARM or x64. Otherwise, undefined.

https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

According to this info, I guess we can omit it in our script.

Reply:
@shiyu1994 `_WIN64` is a "subset" of `_WIN32`. Bolds in `_WIN32` is exactly `_WIN64`:

> `_WIN32` Defined as 1 when the compilation target is 32-bit ARM, **64-bit ARM**, x86, or **x64**. Otherwise, undefined.

> `_WIN64` Defined as 1 when the compilation target is 64-bit ARM or x64. Otherwise, undefined.

https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

Why do you think we need this `ifdef` check?..

Reply:
Sorry, didn't noticed that overlap. I've removed the checks for `_WIN32 ` in configure.win and all checks for `WIN64`. Please check. Thanks!